gsy_framework/data_classes.py

```diff
@@ -487,18 +489,48 @@ def from_dict(cls, bid_offer_match: Dict) -> Optional["BidOfferMatch"]:
     def is_valid_dict(cls, bid_offer_match: Dict) -> bool:
         """Check whether a serialized dict can be a valid BidOfferMatch instance."""
         is_valid = True
-        if len(bid_offer_match.keys()) != len(cls.__annotations__.keys()):
-            is_valid = False
-        elif not all(attribute in bid_offer_match for attribute in cls.__annotations__):
+        required_arguments = (
+            "market_id", "time_slot", "bid", "offer", "selected_energy", "trade_rate")
+        if not all(key in bid_offer_match for key in required_arguments):
             is_valid = False
         elif not isinstance(bid_offer_match["market_id"], str):
             is_valid = False
         elif not isinstance(bid_offer_match["selected_energy"], (int, float)):
             is_valid = False
         elif not isinstance(bid_offer_match["trade_rate"], (int, float)):
             is_valid = False
+        elif not (bid_offer_match.get("matching_requirements") is None or
+                  isinstance(bid_offer_match.get("matching_requirements"), Dict)):
+            is_valid = False
         return is_valid
 
+    @property
+    def bid_energy(self):
```

Comment:
@spyrostz can you check this out, if this approach is okay I will move on with it

Reply:
@Hassan754 looks good to me , and it is flexible enough to add more keys in the table if such a need arises in the future!